ssp/handler.go

```diff
@@ -25,55 +26,61 @@ func (s *Server) Handler() http.Handler {
 	return r
 }
 
-func (server *Server) NutHandler() http.HandlerFunc {
-	type nutResponse struct {
-		Nut string `json:"nut"`
-	}
+// Handler for the nut endpoint
+// Reference: https://www.grc.com/sqrl/sspapi.htm
+// TODO does not yet handle params 0-9, sin or ask
+func (server *Server) NutHandler(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/x-www-form-urlencoded")
 
-	return func(w http.ResponseWriter, r *http.Request) {
-		w.Header().Set("Content-Type", "application/json")
+	nut := server.Nut(clientID(r))
+	server.logger.Printf("Generated nut: %s", nut)
 
-		res := nutResponse{
-			Nut: server.Nut(clientID(r)).String(),
-		}
-		server.logger.Printf("Generated nut: %s", res.Nut)
+	formValues := make(url.Values)
+	formValues.Add("nut", string(nut))
+	formValues.Add("can", sqrl.Base64.EncodeToString([]byte(r.Header.Get("Referer"))))
 
-		if err := json.NewEncoder(w).Encode(res); err != nil {
-			server.logger.Printf("Nut write unsuccessful: %v", err)
-		}
+	if _, err := w.Write([]byte(formValues.Encode())); err != nil {
+		server.logger.Printf("Nut write unsuccessful: %v", err)
 	}
 }
 
-func (server *Server) QRCodeHandler() http.HandlerFunc {
-	return func(w http.ResponseWriter, r *http.Request) {
-		query := r.URL.Query()
-		nut := query.Get("nut")
-		size := atoiWithDefault(query.Get("size"), 256)
-
-		if nut == "" {
-			server.logger.Printf("QR code requested with empty 'nut' parameter")
-			w.WriteHeader(http.StatusNotFound) // TODO: default pending image
-			return
-		}
-		if r.Host == "" {
-			server.logger.Printf("QR code requested with no 'Host' header set")
-			w.WriteHeader(http.StatusNotFound) // TODO: default error image
-			return
-		}
-
-		loginURL := fmt.Sprintf("sqrl://%s/sqrl?nut=%s", requestDomain(r), nut)
-		bytes, err := qrcode.Encode(loginURL, qrcode.Medium, size)
-		if err != nil {
-			server.logger.Printf("Failed to encode login URL '%s': %v", loginURL, err)
-			w.WriteHeader(http.StatusNotFound) // TODO: default error image
-			return
-		}
-
-		w.Header().Set("Content-Type", "image/png")
-		w.WriteHeader(http.StatusOK)
-		if _, err := w.Write(bytes); err != nil {
-			server.logger.Printf("QR code write unsuccessful: %v", err)
-		}
+func (server *Server) QRCodeHandler(w http.ResponseWriter, r *http.Request) {
+	query := r.URL.Query()
+	nut := query.Get("nut")
+	size := atoiWithDefault(query.Get("size"), 256)
+
+	if nut == "" {
+		server.logger.Printf("QR code requested with empty 'nut' parameter")
+		w.WriteHeader(http.StatusBadRequest) // TODO: default pending image
+		w.Write([]byte("Missing nut param"))
```

Comment:
Lines 55 and 61 are failing linting because the errors are unchecked. I'm fine leaving errors unchecked in this case and would suggest the following workaround:

```suggestion
		_, _ = w.Write([]byte("Missing nut param"))
```

Reply:
pushed. makes me a bit sad though... isn't lint supposed to make your code cleaner not messier?